### DISCUSSION TITLE
How to determine original FW without app?

### DISCUSSION THREAD

**Discussion by jklawiter** [roleoroleo/yi-hack-Allwinner-v2#656] [Q&A] [answered]
I understand how to set WiFi without using original app. It's easy, but have to know what's the original firmware if I have Yi home 1080p RFUSY… there are two versions of hack 211ga for 12.0.37* and 291ga for 12.0.35* which is the correct one? :>

**Comment by roleoroleo** [comment, marked as answer]
If the cam is hacked, you can read the file /home/homever

If the cam is not hacked, you have to make a backup copy of the flash, decompress the home partition and read the file.

**Reply by jklawiter** [reply]
Easy! :)

$ cat _mtdblock3.bin.extracted/squashfs-root/homever 
9.0.35.00_202111291330

But maybe (if possible) there should be a tool (little script) for the camera to do that without computer?  

Anyway thank you very much!

**Reply by roleoroleo** [reply]
When the cam is hacked or not?

**Reply by jklawiter** [reply]
I've extracted it from backup using binwalk. But maybe it is possible to get this info with backup script? The hack is getting it from somewhere…

**Reply by roleoroleo** [reply]
The filesystem is comppressed (squash). We need a tool to decompress it: unsquashfs, binwalk, etc...
It's not easy to create a multi-platform tool.

**Reply by roleoroleo** [reply]
But if you are referring to the backup procedure, a dump of the file is already present:
https://github.com/roleoroleo/yi-hack-Allwinner-v2/blob/10277c2dc60eb807241f49120548fff8e93abeb9/sdhack/h52ga/Factory/config.sh#L19

**Reply by jklawiter** [reply]
So if camera is not hacked adding the line  above to this: https://github.com/roleoroleo/yi-hack-Allwinner-v2/wiki/Dump-your-backup-firmware-(SD-card) will not help? Ok if it's impossible it is impossible. 

Thank you!

**Reply by roleoroleo** [reply]
No, it's possible.
Actually is not implemented in the script of the wiki.
But is already present in the hack procedure.
